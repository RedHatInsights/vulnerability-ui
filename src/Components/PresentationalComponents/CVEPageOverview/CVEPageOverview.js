import {
    Card,
    CardBody,
    Grid,
    GridItem,
    Level,
    LevelItem,
    Split,
    SplitItem,
    Stack,
    StackItem,
    Title
} from '@patternfly/react-core';
import { routerParams } from '@red-hat-insights/insights-frontend-components';
import propTypes from 'prop-types';
import React from 'react';
import WithLoader from '../WithLoader/WithLoader';
import './CVEPageOverview.scss';

class CVEPageOverview extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <React.Fragment>
                <GridItem>
                    <Card>
                        <CardBody>
                            <Split>
                                <SplitItem isMain={true} className={'cve-header'}>
                                    <WithLoader loading={this.props.cve.isLoading}>
                                        <Grid>
                                            <GridItem span={1}>{this.props.cve.data.impact}</GridItem>
                                            <GridItem span={11}>
                                                <Title size="xl">{this.props.match.params.cve}</Title>
                                            </GridItem>
                                            <GridItem span={1}>{}</GridItem>
                                            <GridItem span={11}>
                                                <Title size="md">CVSS Base Score: {this.props.cve.data.cvss3_score}</Title>
                                            </GridItem>
                                        </Grid>
                                    </WithLoader>
                                </SplitItem>
                                <SplitItem>
                                    <Title size="md">Publish date: {this.props.cve.data.public_date}</Title>
                                </SplitItem>
                            </Split>
                        </CardBody>
                    </Card>
                </GridItem>
                <Card className="cve-overview-card">
                    <CardBody>
                        <Grid>
                            <GridItem span={8} offset={2}>
                                <Level style={{ textAlign: 'center' }}>
                                    <LevelItem>
                                        <WithLoader loading={this.props.cve.isLoading} variant="cvePageOverviewItem">
                                            <Stack>
                                                <StackItem>{this.props.cve.data.impact}</StackItem>
                                                <StackItem>Impact</StackItem>
                                            </Stack>
                                        </WithLoader>
                                    </LevelItem>
                                    <LevelItem>
                                        <WithLoader loading={this.props.cve.isLoading} variant="cvePageOverviewItem">
                                            <Stack>
                                                <StackItem>{this.props.cve.data.public_date}</StackItem>
                                                <StackItem>Public Date</StackItem>
                                            </Stack>
                                        </WithLoader>
                                    </LevelItem>
                                    <LevelItem>
                                        <WithLoader loading={this.props.cve.isLoading} variant="cvePageOverviewItem">
                                            <Stack>
                                                <StackItem>
                                                    {this.props.cve.errata_list === undefined
                                                        ? 0
                                                        : this.props.cve.data.errata_list.length}
                                                </StackItem>
                                                <StackItem>RHSA(s)</StackItem>
                                            </Stack>
                                        </WithLoader>
                                    </LevelItem>
                                    <LevelItem>
                                        <WithLoader loading={this.props.cve.isLoading} variant="cvePageOverviewItem">
                                            <Stack>
                                                <StackItem>{this.props.cve.data.url_list}</StackItem>
                                                <StackItem>Reference</StackItem>
                                            </Stack>
                                        </WithLoader>
                                    </LevelItem>
                                </Level>
                            </GridItem>
                        </Grid>
                    </CardBody>
                </Card>
            </React.Fragment>
        );
    }
}

CVEPageOverview.propTypes = {
    cve: propTypes.object,
    match: propTypes.object
};

export default routerParams(CVEPageOverview);
