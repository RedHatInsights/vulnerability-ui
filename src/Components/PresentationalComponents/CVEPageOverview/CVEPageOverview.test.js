import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { CVEPageOverview } from './CVEPageOverview';

const cve = {
    data: {
        systems_affected: 2,
        high_visibility: false,
        synopsis: 'CVE-2016-0800',
        public_date: '1970-01-01 00:00:00',
        impact: 'Important',
        description: 'A padding oracle flaw was found in the...',
        cvss2_score: '5.8',
        cvss3_score: null
    },
    meta: {},
    isLoading: false
};

const match = { params: { cve: 'CVE-2016-0800' } };

describe('CVEPageOverview', () => {
    it('should render without data', () => {
        const data = {
            data: {},
            meta: {},
            isLoading: false
        };
        const wrapper = shallow(<CVEPageOverview cve={data} match={match} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with data', () => {
        const wrapper = shallow(<CVEPageOverview cve={cve} match={match} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with enabled WithLoader', () => {
        const data = {
            data: {},
            meta: {},
            isLoading: true
        };
        const wrapper = shallow(<CVEPageOverview cve={data} match={match} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
