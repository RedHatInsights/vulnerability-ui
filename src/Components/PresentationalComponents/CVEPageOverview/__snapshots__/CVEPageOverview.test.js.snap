// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CVEPageOverview should render with data 1`] = `
<Fragment>
  <GridItem
    className=""
    lg={null}
    lgOffset={null}
    lgRowSpan={null}
    md={null}
    mdOffset={null}
    mdRowSpan={null}
    offset={null}
    rowSpan={null}
    sm={null}
    smOffset={null}
    smRowSpan={null}
    span={null}
    xl={null}
    xlOffset={null}
    xlRowSpan={null}
  >
    <Card
      className=""
      component="article"
      isHoverable={false}
    >
      <CardBody
        className=""
        component="div"
        isFilled={true}
      >
        <Split
          className=""
          component="div"
          gutter={null}
        >
          <SplitItem
            className="cve-header"
            isFilled={false}
            isMain={true}
          >
            <WithLoader
              loading={false}
            >
              <Grid
                className=""
                gutter={null}
                lg={null}
                md={null}
                sm={null}
                span={null}
                xl={null}
              >
                <GridItem
                  className=""
                  lg={null}
                  lgOffset={null}
                  lgRowSpan={null}
                  md={null}
                  mdOffset={null}
                  mdRowSpan={null}
                  offset={null}
                  rowSpan={null}
                  sm={null}
                  smOffset={null}
                  smRowSpan={null}
                  span={1}
                  xl={null}
                  xlOffset={null}
                  xlRowSpan={null}
                >
                  Important
                </GridItem>
                <GridItem
                  className=""
                  lg={null}
                  lgOffset={null}
                  lgRowSpan={null}
                  md={null}
                  mdOffset={null}
                  mdRowSpan={null}
                  offset={null}
                  rowSpan={null}
                  sm={null}
                  smOffset={null}
                  smRowSpan={null}
                  span={11}
                  xl={null}
                  xlOffset={null}
                  xlRowSpan={null}
                >
                  <Title
                    size="xl"
                  >
                    CVE-2016-0800
                  </Title>
                </GridItem>
                <GridItem
                  className=""
                  lg={null}
                  lgOffset={null}
                  lgRowSpan={null}
                  md={null}
                  mdOffset={null}
                  mdRowSpan={null}
                  offset={null}
                  rowSpan={null}
                  sm={null}
                  smOffset={null}
                  smRowSpan={null}
                  span={1}
                  xl={null}
                  xlOffset={null}
                  xlRowSpan={null}
                />
                <GridItem
                  className=""
                  lg={null}
                  lgOffset={null}
                  lgRowSpan={null}
                  md={null}
                  mdOffset={null}
                  mdRowSpan={null}
                  offset={null}
                  rowSpan={null}
                  sm={null}
                  smOffset={null}
                  smRowSpan={null}
                  span={11}
                  xl={null}
                  xlOffset={null}
                  xlRowSpan={null}
                >
                  <Title
                    size="md"
                  >
                    CVSS Base Score: 
                  </Title>
                </GridItem>
              </Grid>
            </WithLoader>
          </SplitItem>
          <SplitItem
            className=""
            isFilled={false}
          >
            <Title
              size="md"
            >
              Publish Date: 
              1970-01-01 00:00:00
            </Title>
          </SplitItem>
        </Split>
      </CardBody>
    </Card>
  </GridItem>
  <Card
    className="cve-overview-card"
    component="article"
    isHoverable={false}
  >
    <CardBody
      className=""
      component="div"
      isFilled={true}
    >
      <Grid
        className=""
        gutter={null}
        lg={null}
        md={null}
        sm={null}
        span={null}
        xl={null}
      >
        <GridItem
          className=""
          lg={null}
          lgOffset={null}
          lgRowSpan={null}
          md={null}
          mdOffset={null}
          mdRowSpan={null}
          offset={2}
          rowSpan={null}
          sm={null}
          smOffset={null}
          smRowSpan={null}
          span={8}
          xl={null}
          xlOffset={null}
          xlRowSpan={null}
        >
          <Level
            className=""
            gutter={null}
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            <LevelItem>
              <WithLoader
                loading={false}
                variant="cvePageOverviewItem"
              >
                <Stack
                  className=""
                  component="div"
                  gutter={null}
                >
                  <StackItem
                    className=""
                    isFilled={false}
                  >
                    Important
                  </StackItem>
                  <StackItem
                    className=""
                    isFilled={false}
                  >
                    Impact
                  </StackItem>
                </Stack>
              </WithLoader>
            </LevelItem>
            <LevelItem>
              <WithLoader
                loading={false}
                variant="cvePageOverviewItem"
              >
                <Stack
                  className=""
                  component="div"
                  gutter={null}
                >
                  <StackItem
                    className=""
                    isFilled={false}
                  >
                    1970-01-01 00:00:00
                  </StackItem>
                  <StackItem
                    className=""
                    isFilled={false}
                  >
                    Public Date
                  </StackItem>
                </Stack>
              </WithLoader>
            </LevelItem>
            <LevelItem>
              <WithLoader
                loading={false}
                variant="cvePageOverviewItem"
              >
                <Stack
                  className=""
                  component="div"
                  gutter={null}
                >
                  <StackItem
                    className=""
                    isFilled={false}
                  />
                  <StackItem
                    className=""
                    isFilled={false}
                  >
                    Reference
                  </StackItem>
                </Stack>
              </WithLoader>
            </LevelItem>
          </Level>
        </GridItem>
      </Grid>
    </CardBody>
  </Card>
</Fragment>
`;

exports[`CVEPageOverview should render with enabled WithLoader 1`] = `
<Fragment>
  <GridItem
    className=""
    lg={null}
    lgOffset={null}
    lgRowSpan={null}
    md={null}
    mdOffset={null}
    mdRowSpan={null}
    offset={null}
    rowSpan={null}
    sm={null}
    smOffset={null}
    smRowSpan={null}
    span={null}
    xl={null}
    xlOffset={null}
    xlRowSpan={null}
  >
    <Card
      className=""
      component="article"
      isHoverable={false}
    >
      <CardBody
        className=""
        component="div"
        isFilled={true}
      >
        <Split
          className=""
          component="div"
          gutter={null}
        >
          <SplitItem
            className="cve-header"
            isFilled={false}
            isMain={true}
          >
            <WithLoader
              loading={true}
            >
              <Grid
                className=""
                gutter={null}
                lg={null}
                md={null}
                sm={null}
                span={null}
                xl={null}
              >
                <GridItem
                  className=""
                  lg={null}
                  lgOffset={null}
                  lgRowSpan={null}
                  md={null}
                  mdOffset={null}
                  mdRowSpan={null}
                  offset={null}
                  rowSpan={null}
                  sm={null}
                  smOffset={null}
                  smRowSpan={null}
                  span={1}
                  xl={null}
                  xlOffset={null}
                  xlRowSpan={null}
                />
                <GridItem
                  className=""
                  lg={null}
                  lgOffset={null}
                  lgRowSpan={null}
                  md={null}
                  mdOffset={null}
                  mdRowSpan={null}
                  offset={null}
                  rowSpan={null}
                  sm={null}
                  smOffset={null}
                  smRowSpan={null}
                  span={11}
                  xl={null}
                  xlOffset={null}
                  xlRowSpan={null}
                >
                  <Title
                    size="xl"
                  >
                    CVE-2016-0800
                  </Title>
                </GridItem>
                <GridItem
                  className=""
                  lg={null}
                  lgOffset={null}
                  lgRowSpan={null}
                  md={null}
                  mdOffset={null}
                  mdRowSpan={null}
                  offset={null}
                  rowSpan={null}
                  sm={null}
                  smOffset={null}
                  smRowSpan={null}
                  span={1}
                  xl={null}
                  xlOffset={null}
                  xlRowSpan={null}
                />
                <GridItem
                  className=""
                  lg={null}
                  lgOffset={null}
                  lgRowSpan={null}
                  md={null}
                  mdOffset={null}
                  mdRowSpan={null}
                  offset={null}
                  rowSpan={null}
                  sm={null}
                  smOffset={null}
                  smRowSpan={null}
                  span={11}
                  xl={null}
                  xlOffset={null}
                  xlRowSpan={null}
                >
                  <Title
                    size="md"
                  >
                    CVSS Base Score: 
                  </Title>
                </GridItem>
              </Grid>
            </WithLoader>
          </SplitItem>
          <SplitItem
            className=""
            isFilled={false}
          >
            <Title
              size="md"
            >
              Publish Date: 
            </Title>
          </SplitItem>
        </Split>
      </CardBody>
    </Card>
  </GridItem>
  <Card
    className="cve-overview-card"
    component="article"
    isHoverable={false}
  >
    <CardBody
      className=""
      component="div"
      isFilled={true}
    >
      <Grid
        className=""
        gutter={null}
        lg={null}
        md={null}
        sm={null}
        span={null}
        xl={null}
      >
        <GridItem
          className=""
          lg={null}
          lgOffset={null}
          lgRowSpan={null}
          md={null}
          mdOffset={null}
          mdRowSpan={null}
          offset={2}
          rowSpan={null}
          sm={null}
          smOffset={null}
          smRowSpan={null}
          span={8}
          xl={null}
          xlOffset={null}
          xlRowSpan={null}
        >
          <Level
            className=""
            gutter={null}
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            <LevelItem>
              <WithLoader
                loading={true}
                variant="cvePageOverviewItem"
              >
                <Stack
                  className=""
                  component="div"
                  gutter={null}
                >
                  <StackItem
                    className=""
                    isFilled={false}
                  />
                  <StackItem
                    className=""
                    isFilled={false}
                  >
                    Impact
                  </StackItem>
                </Stack>
              </WithLoader>
            </LevelItem>
            <LevelItem>
              <WithLoader
                loading={true}
                variant="cvePageOverviewItem"
              >
                <Stack
                  className=""
                  component="div"
                  gutter={null}
                >
                  <StackItem
                    className=""
                    isFilled={false}
                  />
                  <StackItem
                    className=""
                    isFilled={false}
                  >
                    Public Date
                  </StackItem>
                </Stack>
              </WithLoader>
            </LevelItem>
            <LevelItem>
              <WithLoader
                loading={true}
                variant="cvePageOverviewItem"
              >
                <Stack
                  className=""
                  component="div"
                  gutter={null}
                >
                  <StackItem
                    className=""
                    isFilled={false}
                  />
                  <StackItem
                    className=""
                    isFilled={false}
                  >
                    Reference
                  </StackItem>
                </Stack>
              </WithLoader>
            </LevelItem>
          </Level>
        </GridItem>
      </Grid>
    </CardBody>
  </Card>
</Fragment>
`;

exports[`CVEPageOverview should render without data 1`] = `
<Fragment>
  <GridItem
    className=""
    lg={null}
    lgOffset={null}
    lgRowSpan={null}
    md={null}
    mdOffset={null}
    mdRowSpan={null}
    offset={null}
    rowSpan={null}
    sm={null}
    smOffset={null}
    smRowSpan={null}
    span={null}
    xl={null}
    xlOffset={null}
    xlRowSpan={null}
  >
    <Card
      className=""
      component="article"
      isHoverable={false}
    >
      <CardBody
        className=""
        component="div"
        isFilled={true}
      >
        <Split
          className=""
          component="div"
          gutter={null}
        >
          <SplitItem
            className="cve-header"
            isFilled={false}
            isMain={true}
          >
            <WithLoader
              loading={false}
            >
              <Grid
                className=""
                gutter={null}
                lg={null}
                md={null}
                sm={null}
                span={null}
                xl={null}
              >
                <GridItem
                  className=""
                  lg={null}
                  lgOffset={null}
                  lgRowSpan={null}
                  md={null}
                  mdOffset={null}
                  mdRowSpan={null}
                  offset={null}
                  rowSpan={null}
                  sm={null}
                  smOffset={null}
                  smRowSpan={null}
                  span={1}
                  xl={null}
                  xlOffset={null}
                  xlRowSpan={null}
                />
                <GridItem
                  className=""
                  lg={null}
                  lgOffset={null}
                  lgRowSpan={null}
                  md={null}
                  mdOffset={null}
                  mdRowSpan={null}
                  offset={null}
                  rowSpan={null}
                  sm={null}
                  smOffset={null}
                  smRowSpan={null}
                  span={11}
                  xl={null}
                  xlOffset={null}
                  xlRowSpan={null}
                >
                  <Title
                    size="xl"
                  >
                    CVE-2016-0800
                  </Title>
                </GridItem>
                <GridItem
                  className=""
                  lg={null}
                  lgOffset={null}
                  lgRowSpan={null}
                  md={null}
                  mdOffset={null}
                  mdRowSpan={null}
                  offset={null}
                  rowSpan={null}
                  sm={null}
                  smOffset={null}
                  smRowSpan={null}
                  span={1}
                  xl={null}
                  xlOffset={null}
                  xlRowSpan={null}
                />
                <GridItem
                  className=""
                  lg={null}
                  lgOffset={null}
                  lgRowSpan={null}
                  md={null}
                  mdOffset={null}
                  mdRowSpan={null}
                  offset={null}
                  rowSpan={null}
                  sm={null}
                  smOffset={null}
                  smRowSpan={null}
                  span={11}
                  xl={null}
                  xlOffset={null}
                  xlRowSpan={null}
                >
                  <Title
                    size="md"
                  >
                    CVSS Base Score: 
                  </Title>
                </GridItem>
              </Grid>
            </WithLoader>
          </SplitItem>
          <SplitItem
            className=""
            isFilled={false}
          >
            <Title
              size="md"
            >
              Publish Date: 
            </Title>
          </SplitItem>
        </Split>
      </CardBody>
    </Card>
  </GridItem>
  <Card
    className="cve-overview-card"
    component="article"
    isHoverable={false}
  >
    <CardBody
      className=""
      component="div"
      isFilled={true}
    >
      <Grid
        className=""
        gutter={null}
        lg={null}
        md={null}
        sm={null}
        span={null}
        xl={null}
      >
        <GridItem
          className=""
          lg={null}
          lgOffset={null}
          lgRowSpan={null}
          md={null}
          mdOffset={null}
          mdRowSpan={null}
          offset={2}
          rowSpan={null}
          sm={null}
          smOffset={null}
          smRowSpan={null}
          span={8}
          xl={null}
          xlOffset={null}
          xlRowSpan={null}
        >
          <Level
            className=""
            gutter={null}
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            <LevelItem>
              <WithLoader
                loading={false}
                variant="cvePageOverviewItem"
              >
                <Stack
                  className=""
                  component="div"
                  gutter={null}
                >
                  <StackItem
                    className=""
                    isFilled={false}
                  />
                  <StackItem
                    className=""
                    isFilled={false}
                  >
                    Impact
                  </StackItem>
                </Stack>
              </WithLoader>
            </LevelItem>
            <LevelItem>
              <WithLoader
                loading={false}
                variant="cvePageOverviewItem"
              >
                <Stack
                  className=""
                  component="div"
                  gutter={null}
                >
                  <StackItem
                    className=""
                    isFilled={false}
                  />
                  <StackItem
                    className=""
                    isFilled={false}
                  >
                    Public Date
                  </StackItem>
                </Stack>
              </WithLoader>
            </LevelItem>
            <LevelItem>
              <WithLoader
                loading={false}
                variant="cvePageOverviewItem"
              >
                <Stack
                  className=""
                  component="div"
                  gutter={null}
                >
                  <StackItem
                    className=""
                    isFilled={false}
                  />
                  <StackItem
                    className=""
                    isFilled={false}
                  >
                    Reference
                  </StackItem>
                </Stack>
              </WithLoader>
            </LevelItem>
          </Level>
        </GridItem>
      </Grid>
    </CardBody>
  </Card>
</Fragment>
`;
